cinema/views.py

```diff
@@ -24,28 +27,60 @@
 class GenreViewSet(viewsets.ModelViewSet):
     queryset = Genre.objects.all()
     serializer_class = GenreSerializer
+    authentication_classes = (TokenAuthentication,)
+
+    def get_permissions(self) -> list:
+        if self.action in ("list", "create"):
+            permission_classes = [IsAdminOrIfAuthenticatedReadOnly]
+        else:
+            raise NotFound()
+        return [permission() for permission in permission_classes]
 
 
 class ActorViewSet(viewsets.ModelViewSet):
     queryset = Actor.objects.all()
     serializer_class = ActorSerializer
+    authentication_classes = (TokenAuthentication,)
+
+    def get_permissions(self) -> list:
+        if self.action in ("list", "create"):
+            permission_classes = [IsAdminOrIfAuthenticatedReadOnly]
+        else:
+            raise NotFound()
+        return [permission() for permission in permission_classes]
 
 
 class CinemaHallViewSet(viewsets.ModelViewSet):
     queryset = CinemaHall.objects.all()
     serializer_class = CinemaHallSerializer
+    authentication_classes = (TokenAuthentication,)
+
+    def get_permissions(self) -> list:
+        if self.action in ("list", "create"):
+            permission_classes = [IsAdminOrIfAuthenticatedReadOnly]
+        else:
+            raise NotFound()
```

Comment:
In `CinemaHallViewSet`, the `get_permissions` method raises a `NotFound` exception for unsupported actions. Consider using `MethodNotAllowed` instead to provide a more accurate error response.